### DISCUSSION TITLE
Multiple keys not working for queryClient.refetchQueries/invalidateQueries for react native app

### DISCUSSION THREAD

**Discussion by amie-wilt** [TanStack/query#2057] [Q&A] [answered]
Hello,

`"react": "^17.0.1"`
`"react-native": "^0.63.4"`
`"react-query": "^3.13.0"`

I'm using this in a react native project and am having some issues with `queryClient.invalidateQueries` and `queryClient.refetchQueries`. I'm looking to refetch multiple queries in a pull-to-refresh action. Both are working with one key, however neither are working with multiple keys. Below I'm only displaying the results for `queryClient.refetchQueries`.

In the first screenshot of my code you can see I'm using one key--`raaReadings`--and it's triggering the refetch as expected, which is demonstrated by the log I put in `node_modules/react-query/lib/core/queryClient.js`, screenshot below.

## Single key working

Single key in `queryClient.refetchQueries`:

![Screen Shot 2021-03-30 at 12 15 02 PM](https://user-images.githubusercontent.com/8226176/113021827-f85aa280-9151-11eb-869d-96e8afb823bd.png)

Console log statement in react-query package:

![Screen Shot 2021-03-30 at 12 18 23 PM](https://user-images.githubusercontent.com/8226176/113021889-09a3af00-9152-11eb-9147-dcc831c148b0.png)

Logged value:

![log-screenshot](https://user-images.githubusercontent.com/8226176/113022281-70c16380-9152-11eb-8a22-b4c8c17e01b0.jpg)

## Multiple keys not working

Multiple keys in `queryClient.refetchQueries`:

![Screen Shot 2021-03-30 at 12 22 51 PM](https://user-images.githubusercontent.com/8226176/113022521-b2eaa500-9152-11eb-82d8-87b3464a32b0.png)

`refetching query` is not logged as it is with a single key and the `isFetching` value for both of the keys I entered is `false`:

![logs-screenshot](https://user-images.githubusercontent.com/8226176/113022849-0ceb6a80-9153-11eb-8f35-20d8d8eef97e.jpg)

I'm not sure if maybe I'm using it incorrectly but I don't see any examples for multiple keys and the answer I found in the Discussions for someone who asked the same question wasn't very insightful:

https://github.com/tannerlinsley/react-query/discussions/302

Thank you and hope you/team is keeping well.

**Comment by TkDodo** [comment, marked as answer]
refetchQueries doesn’t work like that. If you want multiple queries to be refetched that don’t start with the same key, you need to call refetchQueries multiple times.

**Reply by goongoguma** [reply]
I saw interesting behavior of multiple refetchqueries.
As you mentioned above @TkDodo I set two refetchquries to fetch newly updated data after I updateded memo.
```
  queryClient.refetchQueries('useUserInfoMemoAllListById') // fetch all memo data
  queryClient.refetchQueries('useUserInfoMemoMyListById') // fetch the memo data only I wrote
```
This works fine. However when I change the order of these two like below
```
queryClient.refetchQueries('useUserInfoMemoMyListById') // fetch the memo data only I wrote
queryClient.refetchQueries('useUserInfoMemoAllListById') // fetch all memo data
```
react-query fetches only **useUserInfoMemoMyListById** data after I updated the memo. 
Therefore is the order of refetchQuries matter in case multiple refetchQuries are used?

**Reply by TkDodo** [reply]
> Therefore is the order of refetchQuries matter in case multiple refetchQuries are used?

not, it does not. can you post a minimal runnable codesandbox reproduction of the issue?
